src/core/view/GeometryToolView.cpp

```diff
@@ -0,0 +1,43 @@
+#include "GeometryToolView.h"
+
+#include "model/GeometryTool.h"       // for GeometryTool
+#include "model/Stroke.h"             // for Stroke
+#include "util/raii/CairoWrappers.h"  // for CairoSaveGuard
+#include "view/Repaintable.h"         // for Repaintable
+#include "view/View.h"                // for Context
+
+#include "StrokeView.h"  // for StrokeView
+
+using namespace xoj::view;
+
+GeometryToolView::GeometryToolView(const GeometryTool* s, Repaintable* parent): ToolView(parent), s(s) {}
+
+GeometryToolView::~GeometryToolView() = default;
+
+void GeometryToolView::draw(cairo_t* cr) const {
+    xoj::util::CairoSaveGuard saveGuard(cr);
+    this->drawGeometryTool(cr);
```

Comment:
The mask would also need resizing when the tool is rotated, right? So it would only improve performance when the tool or the page is moved. Do you think it is worth the effort?

Reply:
Sorry, it took a while, but I have figured it out now, see commit 45113f950c50791d6e759ebb6b91b3282ee96dfc.
The mask is drawn without (translation and) rotation. So the mask doesn't need redrawing when the geometry tool is rotated (or translated). It is translated and rotated right before `mask.paintTo`. @bhennion Can you take a look at it?

Commit c408a730e323b69e7dce0dafdc7629e2b923c6ff is independent of this PR. I just noticed that `zoomChanged` is fired even when the zoom has not changed at all. That meant that the mask needed resizing/redrawing when pinching to zoom with a scale increase/decrease that was below the threshold set in the preferences.

Reply:
Yes, that's what I had in mind. The code in 45113f9 looks good!

About https://github.com/xournalpp/xournalpp/commit/c408a730e323b69e7dce0dafdc7629e2b923c6ff, it may be worth splitting it to a separate PR, perhaps even destined to 1.1.3.